### DISCUSSION TITLE
Detect duplicate migration keys

### DISCUSSION THREAD

**Discussion by rrrnld** [directus/directus#8596] [Feature Requests]
I just hit a bug where I wasn't able to login using our admin account (which we create in a migration). After some debugging I saw that this was because a migration way later failed due to a duplicate migration key (that was introduced by an automatic merge).

We discussed this in our team and found that it would be nice to include a sanity check in directus when running the migrations. One such fix could be implemented [here](https://github.com/directus/directus/blob/356b96e3e254de44d394e7013ab1d1ae5f9e45d2/api/src/database/migrations/run.ts#L24) along the lines of

``` js
const migrationKeys = new Set(migrations.map(m => m.version))
if (migrations.length > migrationKeys.length) {
  throw new Error('Migration keys collide! Please ensure that every migration uses a unique key.')
}
```

If that sounds good I'll gladly open a pull request. :)

**Comment by rijkvanzanten** [comment]
Totally! Sounds like a very good additional security measure to have in the CLI 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
